daemon/cmd/daemon.go

```diff
@@ -936,6 +934,8 @@ func NewDaemon(ctx context.Context, cleaner *daemonCleanup,
 		case option.Config.EgressMasqueradeInterfaces != "":
 			msg = fmt.Sprintf("BPF masquerade does not allow to specify devices via --%s (use --%s instead).",
 				option.EgressMasqueradeInterfaces, option.Devices)
+		case option.Config.TunnelingEnabled() && !hasFullHostReachableServices():
+			msg = "BPF masquerade requires full host reachable services enabled in tunnel mode."
```

Comment:
This change belongs in the fix commit. You'll have to place the fix commit before the refactoring commit to avoid introducing a regression in between the two commits.

Reply:
yep, make sense